packages/providers/src/providers/localai.ts

```diff
@@ -0,0 +1,24 @@
+import type { ChatProvider, EmbedProvider, ModelProvider, ProviderOptions, ProviderResult } from '../types'
+
+import { generateCRO } from '../utils/generate-cro'
+
+export const createLocalAI = (userOptions?: ProviderOptions<false>):
+  /** @see {@link https://localai.io/model-compatibility/} */
+  ChatProvider<({} & string)>
+  & EmbedProvider<({} & string)>
+  & ModelProvider => {
+  const options: ProviderResult = {
+    ...userOptions,
+    baseURL: userOptions?.baseURL ?? new URL('http://localhost:8080/v1/'),
```

Comment:
For services that don't have unique ports, I think it might be a good idea to provide `OpenAICompatible` instead of creating separate ones for them. As far as I know, llama.cpp also uses this port by default.